### PR TITLE
Port SDL_ShowCursor to SDL3

### PR DESCRIPTION
**Pull request by Starbuck5**

### CODE REVIEW

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: src_c/mouse.c [hunk 1]
```diff
@@ -193,8 +192,18 @@ mouse_set_visible(PyObject *self, PyObject *args)
         }
     }
 
-    toggle = SDL_ShowCursor(toggle);
-    return PyBool_FromLong(toggle);
+    prevstate = PG_CursorVisible();
+
+    // Cursor visibility API also can indicate errors, but we historically
+    // haven't raised those as pygame.errors. Should that change?
```
Comment (ankith26) [inline src_c/mouse.c:198]: Went through both SDL2 and SDL3 source code, and no function actually errors and returns `-1`. Maybe this comment should be along the lines of "according to SDL docs these functions can error with a negative return, but in practice it doesn't happen as of now"?

Comment (Starbuck5) [reply src_c/mouse.c]: Comment updated.

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR :tada:

**Pull request merged**